packages/core/src/browser/shell.ts

```diff
@@ -30,104 +32,109 @@ import { Saveable } from "./saveable";
 import { ContextMenuRenderer } from "./context-menu-renderer";
 import { StatusBarImpl, StatusBarLayoutData } from "./status-bar/status-bar";
 
-export const ApplicationShellOptions = Symbol("ApplicationShellOptions");
-
-/**
- * The class name added to AppShell instances.
- */
+/** The class name added to ApplicationShell instances. */
 const APPLICATION_SHELL_CLASS = 'theia-ApplicationShell';
-
-/**
- * The class name added to side bar instances.
- */
-const SIDEBAR_CLASS = 'theia-SideBar';
-
-/**
- * The class name added to the current widget's title.
- */
+/** The class name added to the main area panel. */
+const MAIN_AREA_CLASS = 'theia-app-main';
+/** The class name added to the main and bottom area panels. */
+const MAIN_BOTTOM_AREA_CLASS = 'theia-app-centers';
+/** The class name added to the left and right area panels. */
+const LEFT_RIGHT_AREA_CLASS = 'theia-app-sides';
+/** The class name added to the current widget's title. */
 const CURRENT_CLASS = 'theia-mod-current';
-
-/**
- * The class name added to the active widget's title.
- */
+/** The class name added to the active widget's title. */
 const ACTIVE_CLASS = 'theia-mod-active';
+/** The class name added to collapsed side panels. */
+const COLLAPSED_CLASS = 'theia-mod-collapsed';
 
 export interface LayoutData {
-    mainArea?: DockLayoutData;
-    leftBar?: SideBarData;
-    rightBar?: SideBarData;
+    mainArea?: DockPanel.ILayoutConfig;
+    leftBar?: SideBarLayoutData;
+    rightBar?: SideBarLayoutData;
+    bottomBar?: SideBarLayoutData;
     statusBar?: StatusBarLayoutData;
 }
 
-export interface SideBarData {
-    activeWidgets?: Widget[];
+export interface SideBarLayoutData {
+    type: 'sidebar',
+    expandedWidgets?: Widget[];
     widgets?: Widget[];
 }
 
-export interface DockLayoutData extends DockPanel.ILayoutConfig {
-    activeWidgets?: Widget[]
-}
+export const MAIN_AREA_TABBAR_CONTEXT_MENU: MenuPath = ['main-area-tabbar-context-menu'];
+export const SIDE_AREA_TABBAR_CONTEXT_MENU: MenuPath = ['side-area-tabbar-context-menu'];
```

Comment:
we should have one context menu for the tabs and remove certain commands by implementing `isVisible`